Simplified/AccountsManager.swift

```diff
@@ -152,10 +158,11 @@ private let prodUrlHash = prodUrl.absoluteString.md5().base64EncodedStringUrlSaf
               NotificationCenter.default.post(name: NSNotification.Name.NYPLCurrentAccountDidChange, object: nil)
               completion(true)
             }
-            if self.currentAccount?.details?.needsAgeCheck ?? false {
+
+            if self.currentAccount?.details?.defaultAuth?.needsAgeCheck ?? false {
```

Comment:
I've added a comment, but generally speaking, there is a logic to handle age check when auth definition is being changed inside NYPLUserAccount. I didn't test it on any coppa auth method, but most probably you will be able to cut out the whole age. check from here, and rely on the one in there. I added a TODO with description in code. Do it before merging this PR, please, as most probably they would cause double check of the age.

Reply:
sounds good, thanks for clarifying this. I filed this ticket for that: https://jira.nypl.org/browse/SIMPLY-2883